interview-questions.md

```diff
@@ -4,65 +4,112 @@ Answer the following questions. First, without external resources. Challenge you
 
 
 
 5. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: a variable that is entered to be put through the logic inside the function.
 
-  Researched answer:
+  Researched answer: The parameters in a function call. In JS the arguments are passed by value: The function only gets to know the values, not the argument's locations. Changes to arguments are not reflected outside of the function.
 
 
 
 6. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: 
+  brief proper pair programming techniques:
+  -clear and open communication
+  -both navigator and driver are engaged in the task, and must understand the how and why
+
+  Roles:
+  -Navigator
+    -Looking miles ahead, macro perspective
+    -will look ahead to see if code will conflict with future tasks
+    -catch simple errors, bugs, and typos
+  
+  -Driver
+    -Looking at what's directly around, micro perspective
+    -using the editor
+    -physically typing the code.
 
   Researched answer:
+  Pair programmers keep each other on task. 
+  Take initiative when their partner is stuck, lowering frustration.
+  Take Breaks!
+
+  "Strong Style" Pairing
+  -The navigator gets to make all the decisions. When the driver needs/wants to contribute an idea, they must relinquish the keyboard. 
+  -the driver does nothing that the reviewer did not direct them to. *"For an idea to go from your head to the computer it **must** go through someone else's hands." -LLewellyn Falco*
+  (the argument for this is that it keeps the navigator engaged.
+  example:
+  "Many times I've observed where the driver says 'just a minute, I've got an idea' and keeps on working, the navigator goes on to check their social media or do some other irrelevant task.")
+
+"Ping Pong" Pairing
+-partner 1 starts off driving until they write a failing test and then hands off the keyboard to partner 2
+-partner 2 makes that test pass, moves on until he writes the next failing test, and transfers the keyboard back to partner 1.
+-repeat this process.
 
 
 
 7. What is something we did in class this week you found helpful?  
 
-  Your answer:
+  Your answer: I thought that learning about github and git was super helpful. It makes sharing work and keeping track of changes in code a lot easier, when I was trying to learn by myself there were only brief explanations on what github was, it was also very difficult if not impossible to completely grasp how to use it without being in a group setting.
+
+  relative to learning by yourself alone, it's much more engaging and it's harder to get bored or get distracted. And having someone there to guide you is very helpful and efficient than just repeatedly running into a wall with your eyes closed until you eventually find the doorway.
```

Comment:
Yay! I'm so glad you are enjoying working in pairs/mobs. It definitely helps keep you on task and interested in the challenge you're working on. 

Learning proper 'git flow' its super crucial to working in a professional environment so I'm super glad you're getting the hang of it and putting importance on it!